### PR TITLE
Prefer trailing dots over leading dots

### PR DESCRIPTION
**Pull request by amiel**
The community styleguide did not pick one and instead encourages teams to pick one.

There is a lengthy discussion here: https://github.com/bbatsov/ruby-style-guide/pull/176 and is a pretty interesting read.

Note that this only applies to ruby.

What are we talking about?
--------------------------

To clarify, here's what we're talking about:

```ruby
  # trailing dots
  one.two.three.
    four

  # leading dots
  one.two.three
    .four
```

My argument
-----------

Having read through the discussion, I propose we use trailing dots.

It seems like the arguments can be oversimplified to mostly ([source](https://github.com/bbatsov/ruby-style-guide/pull/176#issuecomment-24642801):

* leading dots is more visually pleasing
* trailing dots has more functional advantages

Here are some of the arguments I like:

* Basically, when continuing lines like this, you need context on both lines. Using the trailing dot gives context on the first line, indentation gives context on the second line ([source](https://github.com/bbatsov/ruby-style-guide/pull/176#issuecomment-18755860)).
* Code written with leading dots cannot be pasted in to irb/pry ([source](https://github.com/bbatsov/ruby-style-guide/pull/176#issuecomment-20421523)).
* Loading dots are easier for parsers ([source](https://github.com/bbatsov/ruby-style-guide/pull/176#issuecomment-32788590)).
* There are some scenarios in which leading dots can cause bugs ([source](https://github.com/bbatsov/ruby-style-guide/pull/176#issuecomment-32786858)).

### CODE REVIEW

**Comment by amiel** [comment]
I'd love input from everyone on this, but will specifically wait for review from the people that use ruby the most ( @kyle-rader and @nathancarnes ).

**Review by kyle-rader — APPROVED** [review]
@amiel @nathancarnes I agree with trailing dots.
I think the summary of arguments from @amiel is pretty good, but [this guy adds some more points](https://github.com/bbatsov/ruby-style-guide/pull/176#issuecomment-18658186)

Notably, that leading dots don't even work in Ruby 1.8, and there is an actual performance penalty.

**Review by nathancarnes — APPROVED** [review]
Works for me. 

But, more broadly: do not do either except in rare cases. It's probably an indication you have a terrible, terrible Demeter violation.
